Cannot launch local dev server

```
bad runtime process port ['']

panic: initConfig: open config.yaml: no such file or directory

goroutine 1 [running]:
runtime.panic(0x2cf500, 0xc21000ad60)
    /private/var/folders/zj/06c51mbs71z4r22r59jzffjm003chb/T/appengine/go_appengine/goroot/src/pkg/runtime/panic.c:266 +0xb6
main80906.initConfig()
    common.go:28 +0x17a
main80906.initÂ·1()
    server_gae.go:38 +0x5e
main80906.init()
    _extra_imports_6.go:7 +0x15e
main.init()
    _go_main.go:18 +0x3d
```

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -143,7 +143,7 @@ gulp.task('copy-assets', function() {
 });
 
 // Copy backend files.
-gulp.task('copy-backend', function(cb) {
+gulp.task('copy-backend', ['backend:config'], function(cb) {
   gulp.src([
     BACKEND_DIR + '/**/*.go',
     BACKEND_DIR + '/*.yaml',
@@ -243,7 +243,7 @@ gulp.task('pagespeed', pagespeed.bind(null, {
 // Start a standalone server (no GAE SDK needed) serving both front-end and backend,
 // watch for file changes and live-reload when needed.
 // If you don't want file watchers and live-reload, use '--no-watch' option.
-gulp.task('serve', ['backend', 'generate-service-worker-dev'], function() {
+gulp.task('serve', ['backend', 'backend:config', 'generate-service-worker-dev'], function() {
   var noWatch = argv.watch === false;
   var serverAddr = 'localhost:' + (noWatch ? '3000' : '8080');
   var startArgs = ['-d', APP_DIR, '-listen', serverAddr, '-prefix', URL_PREFIX];
@@ -283,7 +283,7 @@ gulp.task('serve', ['backend', 'generate-service-worker-dev'], function() {
 
 // The same as 'serve' task but using GAE dev appserver.
 // If you don't want file watchers and live-reload, use '--no-watch' option.
-gulp.task('serve:gae', ['generate-service-worker-dev'], function(callback) {
+gulp.task('serve:gae', ['backend:config', 'generate-service-worker-dev'], function(callback) {
   var appEnv = process.env.APP_ENV || 'dev';
   var watchFiles = argv.watch !== false;
   var run = startGaeBackend.bind(null, BACKEND_DIR, appEnv, watchFiles, callback);
@@ -318,6 +318,9 @@ gulp.task('copy-experiment-to-site', ['build-experiment'], function(cb) {
 
 // Build self-sufficient backend server binary w/o GAE support.
 gulp.task('backend', buildBackend);
+
+// Create config.yaml if one doesn't exist. Needed to run the server.
+gulp.task('backend:config', generateConfigYaml);
 
 // Backend TDD: watch for changes and run tests in an infinite loop.
 gulp.task('backend:test', function(cb) {
@@ -461,6 +464,19 @@ function changeAppYamlVersion(version, appYamlPath) {
   var appYaml = fs.readFileSync(appYamlPath);
   fs.writeFileSync(appYamlPath, 'version: ' + version + '\n' + appYaml);
   return fs.writeFileSync.bind(fs, appYamlPath, appYaml, null);
+}
+
+// Create dest/config.yaml if it doesn't exist already using config.yaml.template.
+// Needed to start the server.
+function generateConfigYaml(callback) {
+  if (fs.existsSync(BACKEND_DIR + '/config.yaml')) {
+    callback();
+    return;
+  }
+  gulp.src(BACKEND_DIR + '/config.yaml.template', {base: BACKEND_DIR})
+    .pipe($.rename('config.yaml'))
+    .pipe(gulp.dest(BACKEND_DIR))
+    .on('end', callback);
 }
 
 gulp.task('generate-service-worker-dev', ['sass'], function(callback) {

